tensorflow/lite/delegates/nnapi/nnapi_delegate.cc

```diff
@@ -3885,6 +3888,7 @@ TfLiteStatus NNAPIDelegateKernel::BuildGraph(
   std::vector<uint32_t> outputs;
   outputs.reserve(output_tensors->size);
 
+  const auto delegate_options = StatefulNnApiDelegate::GetOptions(delegate);
```

Comment:
As this is a copy and the delegate_options is already obtained in the caller, what about just passing a const reference to StatefulNnApiDelegate::Options instead of 'delegate' pointer?